### PR TITLE
Fix conditional statement determining in-cluster setup

### PR DESCRIPTION
**Pull request by hiddeco**
Follow up on #2741

### CODE REVIEW

File: test/e2e/21_ssh_key_generation.bats [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  kubectl create namespace "$FLUX_NAMESPACE"
+
+  install_flux_with_fluxctl
+}
+
+@test "SSH key is generated" {
+  run fluxctl identity --k8s-fwd-ns "$FLUX_NAMESPACE"
+  [ "$status" -eq 0 ]
+  [ "$output" != "" ]
```
Comment (2opremio) [inline test/e2e/21_ssh_key_generation.bats]: I think we can do a bit better than this. Could parse the output and check it's a valid public key?

Comment (2opremio) [reply test/e2e/21_ssh_key_generation.bats]: Also, we may want to do polling, since the key may not be ready right after flux is deployed.

File: test/e2e/fixtures/kustom/21_ssh_key_generation/generation_patch.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+- op: add
+  path: /spec/template/spec/containers/0/args/-
+  value: --k8s-secret-name=flux-git-deploy-generated
```
Comment (2opremio) [inline test/e2e/fixtures/kustom/21_ssh_key_generation/generation_patch.yaml]: Why is this necessary? I thought the key was generated out of the  out of the box with what's supplied by `fluxctl install`

Comment (2opremio) [reply test/e2e/fixtures/kustom/21_ssh_key_generation/generation_patch.yaml]: Ah, you are doing this to override the key we load in the tests.

Isn't it cleaner to avoid installing that secret?

**Comment by 2opremio** [comment]
Addresses part of #2593

File: test/e2e/fixtures/kustom/21_ssh_key_generation/generation_patch.yaml (hunk 2 above)
Comment (hiddeco) [reply test/e2e/fixtures/kustom/21_ssh_key_generation/generation_patch.yaml]: They were left-overs from a previous version of this, I removed the arguments selecting the folder but forgot to remove the files.

Comment (2opremio) [reply test/e2e/fixtures/kustom/21_ssh_key_generation/generation_patch.yaml]: But, without doing letting flux generate the we won't be testing the big being fixed, I think.

Comment (hiddeco) [reply test/e2e/fixtures/kustom/21_ssh_key_generation/generation_patch.yaml]: Without deploying the git server an empty secret will be applied and Flux will generate the key.

Comment (2opremio) [reply test/e2e/fixtures/kustom/21_ssh_key_generation/generation_patch.yaml]: Ah, good point. Can you add a comment?

File: test/e2e/21_ssh_key_generation.bats (hunk 1 above)
Comment (hiddeco) [reply test/e2e/21_ssh_key_generation.bats]: The key is now validated in two ways:

1. it is parsed by `ssh-keygen` to compute the fingerprint, which will fail for an invalid key
2. the fingerprint is compared to the output of `fluxctl identity -l`
